File: video_1.5_loops/loops.R

FAfiles = list.files("fasta")

nGenes = length(FAfiles)

AAseqs = character(nGenes)
headers = character(nGenes)
#sizes = integer(nGenes)

for(i in 1:nGenes){
  fa = FAfiles[i]
  fullfile = paste0("fasta/", fa)
  mylines = readLines(fullfile)
  aaseq = paste(mylines[-1], collapse = "")
  AAseqs[i] = aaseq
  headers[i] = mylines[1]
  sizes[i] = nchar(aaseq)
}

sizes <- nchar(AAseqs)

AAseqs
headers
sizes

File: video_1.5_loops/loops.py
from os import listdir

FAfiles = listdir('fasta')

#print(FAfiles)

nGenes = len(FAfiles)

AAseqs = ["" for i in range(nGenes)]
headers = ["" for i in range(nGenes)]
sizes = [0 for i in range(nGenes)]

for i in range(nGenes):
    fa = FAfiles[i]
    fullfile = 'fasta/' + fa
    with open(fullfile, mode = 'r') as mycon:
        mylines = mycon.readlines()
    mylines = [line.strip() for line in mylines]
    headers[i] = mylines[0]
    aaseq = "".join(mylines[1:])
    AAseqs[i] = aaseq
    sizes[i] = len(aaseq)

print(AAseqs)
print(headers)
print(sizes)